libraries/rush-lib/src/utilities/prompts/SearchListPrompt.ts

```diff
@@ -0,0 +1,278 @@
+import type { Interface } from 'readline';
+import _ from 'lodash';
+import colors from 'colors/safe';
+
+import type inquirer from 'inquirer';
+import type { Answers, ListQuestion, DistinctChoice } from 'inquirer';
+import BasePrompt from 'inquirer/lib/prompts/base';
+import observe from 'inquirer/lib/utils/events';
+import Paginator from 'inquirer/lib/utils/paginator';
+import type Separator from 'inquirer/lib/objects/separator';
+import type Choice from 'inquirer/lib/objects/choice';
+import type Choices from 'inquirer/lib/objects/choices';
+
+import figures from 'figures';
+
+const { map, takeUntil } = require('rxjs/operators');
+
+interface IKeyPressEvent {
+  key: { name: string; ctrl: boolean; sequence?: string };
+}
+
+export class SearchListPrompt extends BasePrompt<ListQuestion> {
+  protected done!: (result: unknown) => void;
+
+  private readonly _paginator: Paginator;
+  private _selected: number = 0;
+  private _query: string = '';
+  private _firstRender: boolean = true;
+
+  public constructor(question: ListQuestion, readline: Interface, answers: Answers) {
+    super(question, readline, answers);
+
+    if (!this.opt.choices) {
+      this.throwParamError('choices');
+    }
+
+    if (
+      _.isNumber(this.opt.default) &&
+      this.opt.default >= 0 &&
+      this.opt.default < this.opt.choices.realLength
+    ) {
+      this._selected = this.opt.default;
+    } else if (!_.isNumber(this.opt.default) && this.opt.default !== null) {
+      const index: number = _.findIndex(
```

Comment:
`findIndex` is available directly on `Array.prototype`, and if it isn't, update the lib entry in tsconfig.json

Reply:
Removed and replaced